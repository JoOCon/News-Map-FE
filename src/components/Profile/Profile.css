.user-wraper {
  height: -webkit-fill-available;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center; }

.user-profile {
  display: flex;
  flex-direction: column;
  background: rgba(154, 154, 154, 0.9);
  border-radius: .5em;
  padding: 1em;
  align-items: center; }

.user-profile img {
  width: 9em;
  border-radius: .5em; }

.user-profile h1 {
  margin: .5em 0; }

.user-profile h2 {
  margin: .5em 0; }
