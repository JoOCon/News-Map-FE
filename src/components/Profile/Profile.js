import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const Profile = (props) => {
  const {first_name, last_name, email} = props.activeUser;
  return (
    <div className='user-profile'>
      <img alt='the current user profile pic' src='https://www.class-central.com/bundles/classcentralsite/images/icon-programming-and-software-development.png'/>
      <h1>User: {first_name + ' ' + last_name}</h1>
      <h2>Listed Contact: {email}</h2>
    </div>
  );
};

Profile.propTypes = {
  activeUser: PropTypes.object
};

export const mapStateToProps = state => ({
  activeUser: state.activeUser
});

export default connect(mapStateToProps)(Profile);
