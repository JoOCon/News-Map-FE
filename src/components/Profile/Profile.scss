.user-wraper {
  height: 100%;
  width: 100%;
}

.user-profile {
  
}