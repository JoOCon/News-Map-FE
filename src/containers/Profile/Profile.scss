.user-wraper {
  height: -webkit-fill-available;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: radial-gradient(#868686, #52535a);
}

.user-profile {
  display: flex;
  flex-direction: column;
  background: rgba(170, 170, 170, .9);
  border-radius: .5em;
  padding: 1em;
  align-items: center;
}

.user-profile img {
  width: 9em;
  border-radius: .5em;
}

.user-profile h1 {
  margin: .5em 0;
}

.user-profile h2 {
  margin: .5em 0;
}
