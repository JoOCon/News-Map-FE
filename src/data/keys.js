export const googleClientId =
  '241779944002-qa42v6cvk3kpu8da0jimdr364vn41t1r.apps.googleusercontent.com';
