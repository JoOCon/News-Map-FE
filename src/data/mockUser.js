export const mockUser = {
  El: '117977763637801404146',
  Zi: {
    token_type: 'Bearer',
    access_token:
      'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
    scope:
      'openid email profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/plus.me https://www.googleapis.com/auth/userinfo.profile',
    login_hint:
      'AJDLj6JUa8yxXrhHdWRHIV0S13cArCX2skrQ7skgtwzDVxi66KXli5pV9QkfHh_W71S4j789mYsJSJTDHdBzZgGY2XexHHgGyA',
    expires_in: 3600,
    id_token:
      'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
    session_state: { extraQueryParams: { authuser: '0' } },
    first_issued_at: 1540508499764,
    expires_at: 1540512099764,
    idpId: 'google'
  },
  w3: {
    Eea: '117977763637801404146',
    ig: 'Paul K',
    ofa: 'Paul',
    wea: 'K',
    Paa:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/ABtNlbDO0B2Sp9SzOXeqgK19boqsTiIVAA/s96-c/photo.jpg',
    U3: 'pykim0591@gmail.com'
  },
  googleId: '117977763637801404146',
  tokenObj: {
    token_type: 'Bearer',
    access_token:
      'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
    scope:
      'openid email profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/plus.me https://www.googleapis.com/auth/userinfo.profile',
    login_hint:
      'AJDLj6JUa8yxXrhHdWRHIV0S13cArCX2skrQ7skgtwzDVxi66KXli5pV9QkfHh_W71S4j789mYsJSJTDHdBzZgGY2XexHHgGyA',
    expires_in: 3600,
    id_token:
      'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
    session_state: { extraQueryParams: { authuser: '0' } },
    first_issued_at: 1540508499764,
    expires_at: 1540512099764,
    idpId: 'google'
  },
  tokenId:
    'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
  accessToken:
    'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
  profileObj: {
    googleId: '117977763637801404146',
    imageUrl:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/ABtNlbDO0B2Sp9SzOXeqgK19boqsTiIVAA/s96-c/photo.jpg',
    email: 'pykim0591@gmail.com',
    name: 'Paul K',
    givenName: 'Paul',
    familyName: 'K'
  }
};

export const mockStateUser = {
  first_name: 'Paul',
  last_name: 'K',
  email: 'pykim0591@gmail.com',
  gid: '117977763637801404146',
  id: 3,
  created_at: '2018-10-24T23:37:44.724Z',
  updated_at: '2018-10-24T23:37:44.724Z'
};

export const mockUserResponse = {
  created_at: '2018-10-24T23:37:44.724Z',
  email: 'pykim0591@gmail.com',
  first_name: 'Paul',
  gid: '117977763637801404146',
  id: 3,
  last_name: 'K',
  updated_at: '2018-10-24T23:37:44.724Z'
};
