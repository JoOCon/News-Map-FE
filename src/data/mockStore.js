import * as user from './mockUser';
import * as events from './mockEvents';

export const mockStore = {
  activeUser: user.mockStateUser,
  userLocation: { latitude: 39.43243, longitude: -104.98434 },
  watchList: events.userWatchlist,
  events: events.eventsResponse
};
