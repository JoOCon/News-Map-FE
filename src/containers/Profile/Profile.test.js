import React from 'react';
import { Profile, mapStateToProps } from './Profile';
import { mockStore } from '../../data/mockStore';

describe('Profile', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Profile activeUser={mockStore.activeUser} />);
  });

  it('should matchSnapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map to the store properly', () => {
    const expected = mockStore.activeUser;
    const result = mapStateToProps(mockStore);

    expect(result.activeUser).toEqual(expected);
  });
});
