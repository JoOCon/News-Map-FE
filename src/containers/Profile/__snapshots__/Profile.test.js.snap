// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should matchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    activeUser={
      Object {
        "created_at": "2018-10-24T23:37:44.724Z",
        "email": "pykim0591@gmail.com",
        "first_name": "Paul",
        "gid": "117977763637801404146",
        "id": 3,
        "last_name": "K",
        "updated_at": "2018-10-24T23:37:44.724Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="user-profile"
      >
        <img
          alt="the current user profile pic"
        />
        <div>
          <h1>
            User: 
            Paul K
          </h1>
          <h2>
            Listed Contact: 
            pykim0591@gmail.com
          </h2>
        </div>
      </div>,
      "className": "user-wraper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="the current user profile pic"
          />,
          <div>
            <h1>
              User: 
              Paul K
            </h1>
            <h2>
              Listed Contact: 
              pykim0591@gmail.com
            </h2>
          </div>,
        ],
        "className": "user-profile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "the current user profile pic",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                User: 
                Paul K
              </h1>,
              <h2>
                Listed Contact: 
                pykim0591@gmail.com
              </h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "User: ",
                  "Paul K",
                ],
              },
              "ref": null,
              "rendered": Array [
                "User: ",
                "Paul K",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Listed Contact: ",
                  "pykim0591@gmail.com",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Listed Contact: ",
                "pykim0591@gmail.com",
              ],
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="user-profile"
        >
          <img
            alt="the current user profile pic"
          />
          <div>
            <h1>
              User: 
              Paul K
            </h1>
            <h2>
              Listed Contact: 
              pykim0591@gmail.com
            </h2>
          </div>
        </div>,
        "className": "user-wraper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="the current user profile pic"
            />,
            <div>
              <h1>
                User: 
                Paul K
              </h1>
              <h2>
                Listed Contact: 
                pykim0591@gmail.com
              </h2>
            </div>,
          ],
          "className": "user-profile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "the current user profile pic",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  User: 
                  Paul K
                </h1>,
                <h2>
                  Listed Contact: 
                  pykim0591@gmail.com
                </h2>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "User: ",
                    "Paul K",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "User: ",
                  "Paul K",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Listed Contact: ",
                    "pykim0591@gmail.com",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Listed Contact: ",
                  "pykim0591@gmail.com",
                ],
                "type": "h2",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
